haas/model.py

```diff
@@ -122,62 +125,12 @@ class Node(Model):
     project_id    = Column(ForeignKey('project.id'))
     project       = relationship("Project",backref=backref('nodes'))
 
-    # ipmi connection information:
-    ipmi_host = Column(String, nullable=False)
-    ipmi_user = Column(String, nullable=False)
-    ipmi_pass = Column(String, nullable=False)
+    # The Obm info is fetched from the obm class and its respective subclass
+    # pertaining to the node 
+    obm_id = Column(Integer, ForeignKey('obm.id'))
+    obm = relationship("Obm", uselist=False, backref="node")
 
-    def __init__(self, label, ipmi_host, ipmi_user, ipmi_pass):
-        """Register the given node.
-
-        ipmi_* must be supplied to allow the HaaS to do things like reboot
-        the node.
-
-        The node is initially registered with no nics; see the Nic class.
-        """
-        self.label = label
-        self.ipmi_host = ipmi_host
-        self.ipmi_user = ipmi_user
-        self.ipmi_pass = ipmi_pass
-
-    def _ipmitool(self, args):
-        """Invoke ipmitool with the right host/pass etc. for this node.
-
-        `args` - A list of any additional arguments to pass to ipmitool.
-
-        Returns the exit status of ipmitool.
-
-        NOTE: Includes the ``-I lanplus`` flag, available only in IPMI v2+.
-        This is needed for machines which don't accept the older
-        ``lan`` wire protocol.
-        """
-
-        status = call(['ipmitool',
-            '-I', 'lanplus', # See docstring above
-            '-U', self.ipmi_user,
-            '-P', self.ipmi_pass,
-            '-H', self.ipmi_host] + args)
-        if status != 0:
-            logger = logging.getLogger(__name__)
-            logger.info('Nonzero exit status from ipmitool, args = %r', args)
-        return status
-
-    @no_dry_run
-    def power_cycle(self):
-        """Reboot the node via ipmi.
-
-        Returns True if successful, False otherwise.
-        """
-        self._ipmitool(['chassis', 'bootdev', 'pxe'])
-        if self._ipmitool(['chassis', 'power', 'cycle']) == 0:
-            return
-        if self._ipmitool(['chassis', 'power', 'on']) == 0:
-            # power cycle will fail if the machine isn't running, so let's
-            # just turn it on in that case. This way we can save power by
-            # turning things off without breaking the HaaS.
-            return
-        # If it still doesn't work, then it's a real error:
-        raise OBMError('Could not power cycle node %s' % self.label)
+    # The node is initially registered with no nics; see the Nic class.
```

Comment:
Having a wayward comment here doesn't really make sense. I don't think we really need it.

Do the unit tests pass? I feel like at the very least the ones in `tests/unit/model.py` *shouldn't* since they depend on `__init__` having positional arguments. I think removing `__init__` is perfectly sensible, but I suspect there are things that need to be adjusted.